htdocs/website/index.php

```diff
@@ -2155,6 +2155,65 @@
 	}
 }
 
+$form = new Form($db);
+$formadmin = new FormAdmin($db);
+$formwebsite = new FormWebsite($db);
+$formother = new FormOther($db);
+$domainname = '0.0.0.0:8080';
+$tempdir = $conf->website->dir_output.'/'.$websitekey.'/';
+
+// Confirm generation of website sitemaps
+if($action == 'confirmgeneratesitemaps'){
+	$formconfirm = $form->formconfirm($_SERVER["PHP_SELF"].'?website='.$website->ref, $langs->trans('ConfirmSitemapsCreation'), $langs->trans('ConfirmGenerateSitemaps', $object->ref),'generatesitemaps', '', "yes", 1);
+	$action = 'preview';
+}
+
+// Generate web site sitemaps
+if ($action == 'generatesitemaps') {
+	$domtree = new DOMDocument('1.0', 'UTF-8');
+	$root = $domtree->createElementNS('http://www.sitemaps.org/schemas/sitemap/0.9', 'urlset');
+	$domtree->formatOutput = true;
+
+	$sql = "SELECT wp.type_container , wp.pageurl, wp.lang, DATE(wp.tms) as tms, w.virtualhost";
+	$sql .= " FROM ".MAIN_DB_PREFIX."website_page as wp, ".MAIN_DB_PREFIX."website as w";
+	$sql .= " WHERE wp.type_container IN ('page', 'blogpost')";
```

Comment:
Here the request take all page/blog post of all websites. You should add a where to filter on current website;
where w.ref = $websitekey